opentracing/tracecontext.go

```diff
@@ -132,3 +144,14 @@ type TraceContextSource interface {
 	// to NewRootTraceContext().
 	NewRootTraceContext() TraceContext
 }
+
+var kTraceTagRegexp = regexp.MustCompile("^(?i:[a-z0-9][-a-z0-9]*)$")
+
+// Returns the canonicalized version of trace tag key `key`, and true if and
+// only if the key was valid.
+func CanonicalizeTraceTagKey(key string) (string, bool) {
+	if !kTraceTagRegexp.MatchString(key) {
```

Comment:
I think the common mismatch, especially in WSGI environments, would be underscores instead of dashes, which can be easily corrected without panicing.

Another useful trick would be to cache the raw keys into their canonical form, to avoid repetitive regexp evals, since the actual key values are most likely going to be limited in number. I've seen this done in Python [tornado.httutil.HTTPHeaders](http://tornadokevinlee.readthedocs.org/en/latest/_modules/tornado/httputil.html#HTTPHeaders)